api/analyzedphenotypes.cv.api.inc

```diff
@@ -95,196 +102,180 @@ function analyzedphenotypes_cvprop($property = null, $parameter = null) {
+  }
+  elseif ($value == 'name') {
+    // BY CVTERM NAME.
+    $sql_where['field'] = 'cvterm.name';
+    $sql_where['value'] = $identifiers[$value];
+  }
+  elseif ($value == 'keyword') {
+    // BY CVTERM NAME KEYWORD.
+    $sql_where['field'] = 'LOWER(cvterm.name)';
+    $sql_where['match'] = 'LIKE';
+    $sql_where['value'] = '%' . strtolower($identifiers[$value]) . '%';
+  }
+  elseif ($value == 'cvtermcv') {
+    // BY CVTERM NAME AND CV NAME COMBINATION.
+    $sql_where['field'] = 'CONCAT(cvterm.name, \' (\', cv.name, \')\')';
+    $sql_where['value'] = $identifiers[$value];
+  }
+  elseif ($value == 'cv_id') {
+    // BY CV ID NUMBER.
+    $sql_where['field'] = 'cvterm.cv_id';
+    $sql_where['value'] = $identifiers[$value];
   }
 
-  // May request detailed cvterm properties when implied in $options parameter.
+  // field = :value
+  $sql .= ' ' . $sql_where['field'] . ' ' . $sql_where['match'] . ' ' . $sql_where['args'] . ' ';
 
-  // # SEARCH BY NAME.
-  // search is limited to the genus cv configuration.
-  if ($value == 'name') {
-    $args = array(
-      'cv_id' => $sysvar_genus_cv,
-      'name' => $identifiers['name']
+  // WHERE AND.
+  if (isset($identifiers['genus'])) {
+    // Restrict dataset to the cv configuration for a genus.
+    $sysvar = ap_get_variablenames(
+      array('variablename' => $identifiers['genus']),
+      array('set' => 'cvdbon', 'suffix' => 'cv')
     );
 
-    $cvterm = (function_exists('chado_get_cvterm')) ? chado_get_cvterm($args) : tripal_get_cvterm($args);
+    $sysvar_genus_cv = variable_get($sysvar);
+    $sql .= 'AND cvterm.cv_id = ' . $sysvar_genus_cv . ' ';
   }
 
+  // ORDER.
+  $sql .= 'ORDER BY cvterm.name ASC';
 
-  // # SEARCH BY CVTERM ID NUMBER.
-  // search is limited to the genus cv configuration.
-  if ($value == 'cvterm_id') {
-    $args = array(
-      'cv_id' => $sysvar_genus_cv,
-      'cvterm_id' => $identifiers['cvterm_id']
-    );
+  //
+  // EXECUTE.
+  $result = chado_query($sql, array($sql_where['args'] => $sql_where['value']));
 
-    $cvterm = (function_exists('chado_get_cvterm')) ? chado_get_cvterm($args) : tripal_get_cvterm($args);
-  }
+  // RESULT.
+  $cvterm = null;
 
+  if ($result->rowCount()) {
+    if ($options['dataset'] == 'cvtermid' || $options['dataset'] == 'cvtermidname' ) {
+      // Return cvterm_id number.
+      // Return cvterm_id and name.
+      foreach($result as $term) {
+        $d = ($options['dataset'] == 'cvtermid')
+          ? array('cvterm_id' => $term->cvterm_id)
+          : array('cvterm_id' => $term->cvterm_id, 'name' => $term->name);
 
-  // Result options.
+        $cvterm[] = $d;
+      }
+    }
+    elseif ($options['dataset'] == 'namecvname' || $options['dataset'] == 'namecvname_format') {
+      // Return name and cv name - no formatting.
+      // Return name and cv name - with formatting.
+      foreach($result as $term) {
+        $cv = chado_get_cv(array('cv_id' => $term->cv_id));
```

Comment:
You shouldn't need to do this b/c you added cv.name as cv_name to the select clause...

Reply:
This only applies to searching by cvterm name (cv name). Will change the logic.